frontend/class-frontend.php

```diff
@@ -718,7 +718,7 @@ public function robots() {
 		$robots['follow'] = 'follow';
 		$robots['other']  = array();
 
-		if ( ( is_object( $post ) && is_singular() ) || ( WPSEO_Utils::is_woocommerce_active() && is_shop() ) ) {
+		if ( ( is_object( $post ) && is_singular() ) || ( WPSEO_Utils::is_woocommerce_active() && is_shop() && !is_search() ) ) {
```

Comment:
Can you add a space between `!` and `is_search`

Reply:
I've found a better solution for this. We have a WooCommerce shop page class, which determines if a page is a shop page.

You can replace the next line safely
```
( WPSEO_Utils::is_woocommerce_active() && is_shop() && !is_search() )
```

by `$this->woocommerce_shop_page->is_shop_page()`